File: python/get_last_reported_points.py
from client import create_native_client
import nativeconnect.api.v1.nativeconnect_pb2 as native

def main():
    client = create_native_client()

    request = native.GetLastReportedPointsRequest(
        vin="LRW3F7FS9PC893577",
        points=5
    )

    response = client.GetLastReportedPoints(request)

    print(response)

     
if __name__ == "__main__":
    main()

File: typescript/src/get_last_reported_points.ts
import { createNodeClient } from "./client"
import {
  GetLastReportedPointsRequestSchema, GetLastReportedPointsResponseSchema
} from "@buf/nativeconnect_api.bufbuild_es/nativeconnect/api/v1/nativeconnect_pb";
import {create, toJsonString} from "@bufbuild/protobuf";

const client = createNodeClient()


const request = create(GetLastReportedPointsRequestSchema, {
  vin: "LRW3F7FS9PC893577",
  points: 3,
})


const response = await client.getLastReportedPoints(request)

console.log(toJsonString(GetLastReportedPointsResponseSchema, response, { prettySpaces: 2 }))
